static/js/containers/CreatePostPage_test.js

```diff
@@ -414,66 +414,83 @@ describe("CreatePostPage", () => {
+      [LINK_TYPE_ANY, LINK_TYPE_TEXT, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_TEXT, null, false, true],
+      [LINK_TYPE_ANY, LINK_TYPE_LINK, LINK_TYPE_LINK, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_LINK, LINK_TYPE_TEXT, true, true],
+      [LINK_TYPE_ANY, LINK_TYPE_LINK, null, false, true],
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, LINK_TYPE_LINK, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_ANY, LINK_TYPE_ANY, null, false, false],
       // starting with LINK
-      [LINK_TYPE_LINK, LINK_TYPE_TEXT, LINK_TYPE_LINK, true],
-      [LINK_TYPE_LINK, LINK_TYPE_TEXT, null, true],
-      [LINK_TYPE_LINK, LINK_TYPE_ANY, LINK_TYPE_LINK, false],
-      [LINK_TYPE_LINK, LINK_TYPE_ANY, null, false],
+      [LINK_TYPE_LINK, LINK_TYPE_TEXT, LINK_TYPE_LINK, true, true],
+      [LINK_TYPE_LINK, LINK_TYPE_TEXT, null, false, true],
+      [LINK_TYPE_LINK, LINK_TYPE_ANY, LINK_TYPE_LINK, true, false],
+      [LINK_TYPE_LINK, LINK_TYPE_ANY, LINK_TYPE_LINK, false, true],
+      [LINK_TYPE_LINK, LINK_TYPE_ANY, null, false, false],
       // starting with TEXT
-      [LINK_TYPE_TEXT, LINK_TYPE_TEXT, LINK_TYPE_TEXT, false],
-      [LINK_TYPE_TEXT, LINK_TYPE_LINK, LINK_TYPE_TEXT, true],
-      [LINK_TYPE_TEXT, LINK_TYPE_LINK, null, true],
-      [LINK_TYPE_TEXT, LINK_TYPE_ANY, LINK_TYPE_TEXT, false],
-      [LINK_TYPE_TEXT, LINK_TYPE_ANY, null, false]
-    ].forEach(([fromLinkType, toLinkType, formValue, shouldDispatch]) => {
-      it(`${
-        shouldDispatch ? "dispatches" : "doesn't dispatch"
-      } FORM_UPDATE if the postType is ${String(
-        formValue
-      )} when it goes from ${String(fromLinkType)} to ${toLinkType}`, () => {
-        const dispatch = helper.sandbox.stub()
-        currentChannel.link_type = fromLinkType
-        const nextChannel = channels[1]
-        nextChannel.link_type = toLinkType
-        const page = new InnerCreatePostPage()
-        const props: any = {
-          dispatch: dispatch,
-          channel:  nextChannel,
-          postForm: {
-            value: {
-              postType: formValue
+      [LINK_TYPE_TEXT, LINK_TYPE_TEXT, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_TEXT, LINK_TYPE_LINK, LINK_TYPE_TEXT, true, true],
+      [LINK_TYPE_TEXT, LINK_TYPE_LINK, null, false, true],
+      [LINK_TYPE_TEXT, LINK_TYPE_ANY, LINK_TYPE_TEXT, true, false],
+      [LINK_TYPE_TEXT, LINK_TYPE_ANY, LINK_TYPE_TEXT, false, true],
+      [LINK_TYPE_TEXT, LINK_TYPE_ANY, null, false, false]
+    ].forEach(
+      ([fromLinkType, toLinkType, formValue, hasUrlOrText, shouldDispatch]) => {
+        it(`${shouldIf(
+          shouldDispatch
+        )} dispatch FORM_UPDATE if the postType is ${String(
+          formValue
+        )} when it goes from ${String(
+          fromLinkType
+        )} to ${toLinkType} and user has ${
+          hasUrlOrText ? "" : "not "
+        }entered a value`, () => {
```

Comment:
This test case is huge now. If it's clean enough to do so, please create a separate test case for switching from one post type to another

Reply:
Test has been split up.